include/app/tabs/camera.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef CAMERA_HPP_
+#define CAMERA_HPP_
+
+#include <QMediaPlayer>
+#include <QVideoWidget>
+#include <QLabel>
+
+#include <ui_camera_settings.h>
+
+class CameraTab : public QWidget {
+    Q_OBJECT
+
+   public:
+    CameraTab(QWidget *parent = nullptr);
+
+   signals:
+    void media_status_changed( QString );
+
+   public slots:
+    void connect_stream();
+    void disconnect_stream();
+    void on_camName_changed(QString name);
+    void on_toggle_connect();
+
+   private slots:
+    void ChangedStatus(QMediaPlayer::MediaStatus);
+
+   private:

```

Comment:
all class variables should be snake case

Reply:
also in Qt-ness, shouldnt they all be pointers?

Reply:
Is that a Qt thing? It adds code bloat to separately declare and "new" all of the composed objects. Also, dots are nicer than arrows, at least IMO. Is there a reason for using pointers?

Reply:
kinda, its really about the lifetime of your objects.

When you create the object on the stack (not using "new"), it only exists when its in scope. When you dynamically allocate the object, it'll exist until you delete, so you can guarantee it'll be there when you need it.

This would technically be fine since they're class variables and are in scope for the lifetime of the class, but still not ideal.

As long as you "register" it properly with Qt, then you don't need to worry about deleting it or anything because Qt will end up handling that.

Reply:
Also is there really any additional "bloat"? When you create these objects dynamically, you set the parameters explicitly instead of using the `set` methods.

Reply:
Why would you need the widgets on a page have a different lifetime than the class for the page?

Anyway, I'll change it to the pointer way because that's how it's done elsewhere.

Reply:
The bloat I'm referring to (which isn't that bad, just superfluous IMO) is the "new" statements, which are implicit when the objects are members.

Reply:
As a side point, you don't even have to declare them at all when using AUTOUIC

Reply:
well "new" isnt implicit, it creates an entirely different type of variable (static vs dynamic allocation)

and I agree you definitely don't want the widgets to extend past the lifetime of the class, but there may be other side effects of static initialization (big one being running out of space on the stack)